components/clm/src/cpl/lnd_disagg_forc.F90

```diff
@@ -0,0 +1,690 @@
+    type(lnd2atm_type)               , intent(in)    :: lnd2atm_vars
+    integer                    , intent(in)    :: method
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: c,l,fc     ! indices
+    integer  :: nstep
+    real(r8) :: hsurf_t      ! column-level elevation (m)
+    real(r8) :: hsurf_g      ! gridcell-level elevation (m)
+
+    real(r8) :: tair_g         ! original gridcell mean air temperature
+    real(r8) :: tair_t         ! downscaled topounit air temperature
+    real(r8) :: tsfc_g         ! original gridcell surface temperature
+    real(r8) :: tsfc_t         ! downscaled topounit surface temperature
+    real(r8) :: lwrad_g        ! original gridcell mean LW radiation
+    real(r8) :: lwrad_t        ! downscaled topounit LW radiation
+    real(r8) :: newsum_lwrad_g ! weighted sum of column-level lwrad after normalization
+
+    character(len=*), parameter :: subname = 'downscale_longwave_top'
+    !-----------------------------------------------------------------------
+
+    nstep = get_nstep()
+    
+    ! Do the downscaling
+    hsurf_g = grc_pp%elevation(g) !ldomain%topo2(g)
+    hsurf_t = top_pp%elevation(t)
+
+    ! Here we assume that deltaLW = (dLW/dT)*(dT/dz)*deltaz
+    ! We get dLW/dT = 4*eps*sigma*T^3 = 4*LW/T from the Stefan-Boltzmann law,
+    ! evaluated at the mean temp.
+    ! We assume the same temperature lapse rate as above.
+
+    tair_g = x2l(index_x2l_Sa_tbot,i)
+    tair_t = top_as%tbot(t)
+    tsfc_g = lnd2atm_vars%t_rad_grc(g)
+    lwrad_g = x2l(index_x2l_Faxa_lwdn,i)
+    if (method == 0 .or. (method == 1 .and. nstep == 0)) then
+       lwrad_t = lwrad_g - &
+          4.0_r8 * lwrad_g/(0.5_r8*(tair_t+tair_g)) * &
+          lapse_glcmec * (hsurf_t - hsurf_g)
+    else
+       
+       lwrad_t = lwrad_g + lnd2atm_vars%eflx_lwrad_out_grc(g) * &
+          4._r8 * (tair_t - tair_g) / tsfc_g
+!       if (masterproc) then  ! TKT debugging
+!            write(iulog,*) ' lwrad_g =  ', lwrad_g 
+!            write(iulog,*) ' lnd2atm_vars%eflx_lwrad_out_grc(g) ', lnd2atm_vars%eflx_lwrad_out_grc(g) 
+!            write(iulog,*) ' tair_t ',tair_t
+!            write(iulog,*) ' tair_g ',tair_g
+!            write(iulog,*) ' tsfc_g ',tsfc_g
+!            write(iulog,*) ' lwrad_t ',lwrad_t
+!       end if
+       
+    end if
+       top_af%lwrad(t) = lwrad_t
+
+!            newsum_lwrad_g = newsum_lwrad_g + top_pp%wtgcell(t)*lwrad_t
+
+
+       ! Make sure that, after normalization, the grid cell mean is conserved
+!           if (sum_wts_g > 0._r8) then
+!               if (abs((newsum_lwrad_g / sum_wts_g) - lwrad_g) > 1.e-8_r8) then
+!                  write(iulog,*) 'g, newsum_lwrad_g, sum_wts_g, forc_lwrad_g: ', &
+!                       g, newsum_lwrad_g, sum_wts_g, lwrad_g
+!                  call endrun(msg=' ERROR: Energy conservation error downscaling longwave'//&
+!                       errMsg(__FILE__, __LINE__))
+!               end if
+!            end if
```

Comment:
Remove the commented out code

Reply:
Thanks! Commented out code has been removed.